charon/src/translate/translate_traits.rs

```diff
@@ -246,9 +246,12 @@ impl<'tcx, 'ctx> TranslateCtx<'tcx, 'ctx> {
         trace!("About to translate trait decl:\n{:?}", rust_id);
 
         let def_id = self.register_trait_decl_id(&None, rust_id)?;
+
+        let item_meta = self.translate_item_meta_from_rid(rust_id);
+
         // We may need to ignore the trait (happens if the trait is a marker
         // trait like [core::marker::Sized]
-        if def_id.is_none() {
+        if def_id.is_none() || item_meta.opaque {
```

Comment:
This check ignores the trait altogether: you should remove the `item_meta.opaque` from here (we do this for very specific traits that we completely ignore so that they do not appear anywhere in the crate).

We actually have no way of completely ignoring trait declarations at this point: if `item_meta.opaque` is `true` you should print a warning stating that this attribute has no effect on trait declarations and that we will thus ignore (with span, name of the declaration, etc.).

Reply:
We could translate the trait name but not translate its methods. That way it can be used as a bound.

Reply:
Correct, but we often lookup trait methods in other places, so we have to pay attention not to trigger bugs by doing something like this.

Reply:
Fair, we can do that later if we have a use for it